e2e/tests/transfer/forwarding_test.go

```diff
@@ -289,3 +289,78 @@ func (s *TransferForwardingTestSuite) TestChannelUpgradeForwarding_Succeeds() {
 		s.Require().Equal(expected, actualBalance.Int64())
 	})
 }
+
+// TestFailedForwarding tests the scenario in which the packet is sent from
+// A to C (through B) but it can't reach C (we use an invalid address).
+func (s *TransferForwardingTestSuite) TestFailedForwarding() {
+	t := s.T()
+	ctx := context.TODO()
+
+	testName := t.Name()
+	t.Parallel()
+	relayer := s.CreateDefaultPaths(testName)
+	chains := s.GetAllChains()
+
+	chainA, chainB, chainC := chains[0], chains[1], chains[2]
+
+	chainADenom := chainA.Config().Denom
+
+	chainAWallet := s.CreateUserOnChainA(ctx, testvalues.StartingTokenAmount)
+	chainAAddress := chainAWallet.FormattedAddress()
+
+	chainBWallet := s.CreateUserOnChainB(ctx, testvalues.StartingTokenAmount)
+
+	chainCWallet := s.CreateUserOnChainC(ctx, testvalues.StartingTokenAmount)
+
+	channelAtoB := s.GetChainAChannelForTest(testName)
+	channelBtoC := s.GetChannelsForTest(chainB, testName)[1]
+
+	s.Require().NoError(test.WaitForBlocks(ctx, 1, chainA, chainB), "failed to wait for blocks")
+
+	chainBstartingBalance, err := testsuite.GetChainBalanceForDenom(ctx, chainB, chainADenom, chainBWallet)
```

Comment:
Usage of `chainADenom` here and for the balance on chain C is not correct, right? We should be using the IBC denom con chain B and C respectively, right?

Reply:
you're absolutely right, I'll fix this, thanks for catching it!